### PR TITLE
Added Climbing Stairs DP Solution

### PR DESCRIPTION
**Pull request by a7medayman6**
C++ implementation of Climbing stairs problem solution using Dynamic Programming Approach, with user input.
Example test cases, time complexity, and space complexity are comments at the end of the code.
Updated Algorithm/DP/readme.md and added a link to the problem solution.

### CODE REVIEW

**Comment by raksha009** [comment]
Heyy, always create a PR after the issue is assigned to you. Otherwise we'll close both the issue and the PR.
Making an exception for now :)

File: Algorithm/DP/readme.md [hunk 1]
```diff
@@ -42,3 +42,5 @@ be avoided by constructinng a temporary array dp[] and memoizing the computed va
 - Longest Common Subsequence ----> [C++](/Code/C++/longest_common_subsequence.cpp)
 
 - Minimum Number of Deletions To Make a String Palindrome ---->[C++](/Code/C++/min_deletions_to_make_string_palindrome.cpp)
+
```
Comment (raksha009) [inline Algorithm/DP/readme.md:45]: Remove the extra line - line 43 and 45

File: Code/C++/Climbing_Stairs.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/*
+
```
Comment (raksha009) [inline Code/C++/Climbing_Stairs.cpp:2]: Remove the extra line

File: Code/C++/Climbing_Stairs.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+*/
+
+
+#include <iostream>
+using namespace std;
+
+long long ClimbStairs(int N) 
+{
+	/*
+	 * Declare an array to store the steps at each stair
```
Comment (raksha009) [inline Code/C++/Climbing_Stairs.cpp:21]: Create multiline comments via 
/*
*/
Donot add * at every line.

File: Code/C++/Climbing_Stairs.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	if(n > 0)
+	 	cout << "The number of distinct ways can you climb to the top :\t" << ClimbStairs(n) << endl; 
+	else
+		cout << "The number of stairs must be greater than 0." <<endl;
+
+
+  return 0;
+}
+
+/*
```
Comment (raksha009) [inline Code/C++/Climbing_Stairs.cpp:65]: 1. Add all this in a single multiline comment.
2. Add Time Complexity after Test Cases

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by a7medayman6** [comment]
> Heyy, always create a PR after the issue is assigned to you. Otherwise we'll close both the issue and the PR.
> Making an exception for now :)

Sorry for that, I didn't get it first as it wasn't clear about pushing new algorithms, Thank You.

I will make the changes you pointed at, thank you again.

**Comment by a7medayman6** [comment]
@raksha009 
hello, I changed the requested changes under the same PR #1378

File: Code/C++/Climbing_Stairs.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+*/
+
+
+#include <iostream>
+using namespace std;
+
+long long ClimbStairs(int N) 
+{
+	/*
+	 	Base Case : at the first stair (steps[0]) the number of ways to get there is 1
```
Comment (raksha009) [inline Code/C++/Climbing_Stairs.cpp:20]: Add remarks via single line comments in between the code.

**Review by raksha009 — CHANGES_REQUESTED** [review]
Just make this small change., then we are good to go.

File: Code/C++/Climbing_Stairs.cpp (hunk 5 above)
Comment (a7medayman6) [reply Code/C++/Climbing_Stairs.cpp]: @raksha009  done

**Comment by a7medayman6** [comment]
@raksha009  there is a conflict in Algorithm/DP/readme.md should I resolve it ?

**Comment by raksha009** [comment]
> @raksha009 there is a conflict in Algorithm/DP/readme.md should I resolve it ?

yes please

File: Code/C++/Climbing_Stairs.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		s1 = tmp;		
+	}		
+
+	//Return the steps at the final stair.
+	return s2;
+}
+
+int main() 
+{
+
```
Comment (raksha009) [inline Code/C++/Climbing_Stairs.cpp:46]: Remove extra line here

**Review by raksha009 — CHANGES_REQUESTED** [review]
Just make this change and resolve conflicts. Rest LGTM 👍🏼

File: Code/C++/Climbing_Stairs.cpp (hunk 6 above)
Comment (a7medayman6) [reply Code/C++/Climbing_Stairs.cpp]: I'm trying to resolve the conflict but something is wrong, can you resolve it please or tell me what to do ?

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
resolve the conflict.

**Comment by a7medayman6** [comment]
@raksha009  I resolved the conflict.

**Comment by tarun26091999** [comment]
#1379

**Review by tarun26091999 — APPROVED** [review]

**Pull request merged**